src/task.c

```diff
@@ -728,16 +728,16 @@ void NORETURN throw_internal(jl_value_t *e)
     }
     else {
         if (jl_current_task == jl_root_task) {
-            JL_PRINTF(JL_STDERR, "fatal: error thrown and no exception handler available.\n");
+            jl_printf(JL_STDERR, "fatal: error thrown and no exception handler available.\n");
             jl_static_show(JL_STDERR, e);
-            JL_PRINTF(JL_STDERR, "\n");
+            jl_printf(JL_STDERR, "\n");
             jl_exit(1);
         }
         jl_current_task->exception = e;
         finish_task(jl_current_task, e);
         assert(0);
     }
-    jl_exit(1);
+    assert(0);
```

Comment:
ah, right. i forgot jl_longjmp was also marked NORETURN

that's more-or-less just the default behavior for C compilers. there is a small performance hit from having them on everywhere, although I do tend to try to use them (esp. w/ setting `LLVM_ASSERTIONS=1`)